File: truth_set/python/structure/solution-approach/semantic-duplication/services/orderService.py
from datetime import datetime

class OrderService:
    def __init__(self):
        self.orders = []
    
    def add_order(self, product):
        new_order = {
            'order_id': len(self.orders) + 1,
            'product': product,
            'order_date': datetime.now()
        }
        self.orders.append(new_order)
        formatted_date = self.format_date(new_order['order_date'], 'DD-MM-YYYY')
        print(f"Order Added: {new_order['product']}, Order Date: {formatted_date}")
    
    def list_orders(self):
        for order in self.orders:
            formatted_date = self.format_date(order['order_date'])
            print(f"Order ID: {order['order_id']}, Product: {order['product']}, Order Date: {formatted_date}")
    
    def format_date(self, date, format='YYYY-MM-DD'):
        year = date.year
        month = f"{date.month:02d}"
        day = f"{date.day:02d}"
        
        if format == 'MM/DD/YYYY':
            return f"{month}/{day}/{year}"
        elif format == 'DD-MM-YYYY':
            return f"{day}-{month}-{year}"
        else:
            return f"{year}-{month}-{day}"
\ No newline at end of file

File: truth_set/go/structure/solution-approach/semantic-duplication/main/services/orderService.go
package services

import (
	"fmt"
	"time"
)

type Order struct {
	OrderID   int
	Product   string
	OrderDate time.Time
}

type OrderService struct {
	orders []Order
}

func NewOrderService() *OrderService {
	return &OrderService{orders: make([]Order, 0)}
}

func (s *OrderService) AddOrder(product string) {
	newOrder := Order{
		OrderID:   len(s.orders) + 1,
		Product:   product,
		OrderDate: time.Now(),
	}
	s.orders = append(s.orders, newOrder)
	formattedDate := s.formatDate(newOrder.OrderDate, "DD-MM-YYYY")
	fmt.Printf("Order Added: %s, Order Date: %s\n", newOrder.Product, formattedDate)
}

func (s *OrderService) ListOrders() {
	for _, order := range s.orders {
		formattedDate := s.formatDate(order.OrderDate, "YYYY-MM-DD")
		fmt.Printf("Order ID: %d, Product: %s, Order Date: %s\n",
			order.OrderID, order.Product, formattedDate)
	}
}

func (s *OrderService) formatDate(date time.Time, format string) string {
	year := date.Format("2006")
	month := date.Format("01")
	day := date.Format("02")

	switch format {
	case "MM/DD/YYYY":
		return month + "/" + day + "/" + year
	case "DD-MM-YYYY":
		return day + "-" + month + "-" + year
	default:
		return year + "-" + month + "-" + day
	}
}
\ No newline at end of file